tests/multi_ptr/multi_ptr_subscript_op_common.h

```diff
@@ -0,0 +1,168 @@
+  using multi_ptr_t = const sycl::multi_ptr<T, space, decorated>;
+  using arr_t = std::array<T, array_size>;
+
+ public:
+  /**
+   * @param type_name Current data type string representation
+   * @param address_space_name Current sycl::access::address_space string
+   *        representation
+   * @param is_decorated_name Current sycl::access::decorated string
+   *        representation
+   */
+  void operator()(const std::string &type_name,
+                  const std::string &address_space_name,
+                  const std::string &is_decorated_name) {
+    auto queue = sycl_cts::util::get_cts_object::queue();
+    bool ret_type_check;
+
+    // arrays for returned and expected values
+    arr_t ret_arr = multi_ptr_common::init_array<T, array_size>::value;
+    arr_t ret_arr_neg = multi_ptr_common::init_array<T, array_size>::value;
+    arr_t exp_arr = multi_ptr_common::init_array<T, array_size>::value;
+
+    for (size_t i = 0; i < array_size; ++i) {
+      exp_arr[i] = i;
+    }
+
+    {
+      sycl::buffer<bool> ret_type_buffer(&ret_type_check, {1});
+      sycl::buffer<T> ret_arr_buffer(ret_arr.data(), {array_size});
+      sycl::buffer<T> ret_arr_neg_buffer(ret_arr_neg.data(), {array_size});
+      sycl::buffer<T> exp_arr_buffer(exp_arr.data(), {array_size});
+
+      queue.submit([&](sycl::handler &cgh) {
+        auto ret_type_acc =
+            ret_type_buffer.template get_access<sycl::access_mode::write>(cgh);
+        auto ret_arr_acc =
+            ret_arr_buffer.template get_access<sycl::access_mode::write>(cgh);
+        auto ret_arr_neg_acc =
+            ret_arr_neg_buffer.template get_access<sycl::access_mode::write>(
+                cgh);
+
+        auto test_device_code = [=](auto acc_for_multi_ptr,
+                                    auto acc_for_multi_ptr_negative) {
+          multi_ptr_t multi_ptr(acc_for_multi_ptr);
+          multi_ptr_t multi_ptr_negative(acc_for_multi_ptr_negative);
+
+          // standard indexation
+          for (int i = 0; i < array_size; i++) {
+            ret_arr_acc[i] = multi_ptr[i];
+          }
+
+          // negative indexation
+          for (int i = 0; i < array_size; i++) {
+            ret_arr_neg_acc[i] = multi_ptr_negative[-i];
+          }
+
+          // type check
+          ret_type_acc[0] = std::is_same_v<decltype(multi_ptr[0]),
+                                           typename multi_ptr_t::reference>;
+        };
+
+        if constexpr (space == sycl::access::address_space::local_space) {
+          sycl::local_accessor<T> exp_arr_acc{{array_size}, cgh};
+          sycl::local_accessor<T> exp_arr_neg_acc{{array_size}, cgh};
+          cgh.single_task([=] {
+            value_operations::assign(exp_arr_acc, exp_arr);
+            value_operations::assign(exp_arr_neg_acc, exp_arr);
+
+            // pointer to the end of the array
+            T *arr_end = const_cast<T *>(&exp_arr_neg_acc[array_size - 1]);
```

Comment:
It is unclear to me why the `const` happens at the first place.

Reply:
Removed.